core/src/test/java/org/apache/calcite/sql/test/SqlOperatorBaseTest.java

```diff
@@ -9096,6 +9114,12 @@ protected void checkAggType(SqlTester tester, String expr, String type) {
         false);
     final String[] values = {"1", "2", "1"};
     tester.checkAgg("bit_xor(x)", values, 2, 0);
+    final String[] binaryValues = {
+        "CAST(x'01' AS BINARY)",
+        "cast(x'02' as BINARY)",
+        "cast(x'01' AS BINARY)",
+        "cast(null AS BINARY)"};
+    tester.checkAgg("bit_xor(x)", binaryValues, "02", 0);
```

Comment:
What if it has only 1 element?

Reply:
Can you add the case?

Reply:
Yes, I have added. When it has only 1 element, the result is itself.

Reply:
Can you add a case with duplicate element and distinct for BIT_XOR? I am afraid it won't work.

Reply:
You are right. Distinct for BIT_XOR isn't work. For SqlBitOpAggFunction#getDistinctOptionality is Optionality.IGNORED. So I think, for  BIT_AND and BIT_OR, We can set Optionality.IGNORED. And for BIT_XOR, We can set Optionality.OPTIONAL. What do you think of it?

Reply:
Correct.